staging/src/k8s.io/kubectl/pkg/drain/drain.go

```diff
@@ -220,6 +220,8 @@ func filterPods(podList *corev1.PodList, filters []PodFilter) *PodDeleteList {
 		// Add the pod to PodDeleteList no matter what PodDeleteStatus is,
 		// those pods whose PodDeleteStatus is false like DaemonSet will
 		// be catched by list.errors()
+		pod.Kind = "Pod"
+		pod.APIVersion = "v1"
```

Comment:
Is there a better way to add this? The reason it is not present is that these objects were loaded from a PodList, which does not include type meta on its items in the API response.